internal/provider/resource/app_connection/app_connection_gitlab.go

```diff
@@ -0,0 +1,135 @@
+package resource
+
+import (
+	"context"
+	infisical "terraform-provider-infisical/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+type AppConnectionGitlabCredentialsModel struct {
+	AccessToken     types.String `tfsdk:"access_token"`
+	InstanceUrl     types.String `tfsdk:"instance_url"`
+	AccessTokenType types.String `tfsdk:"access_token_type"`
+}
+
+const AppConnectionGitlabAuthMethodAccessToken = "access-token"
+
+func NewAppConnectionGitlabResource() resource.Resource {
+	return &AppConnectionBaseResource{
+		App:               infisical.AppConnectionAppGitlab,
+		AppConnectionName: "GitLab",
+		ResourceTypeName:  "_app_connection_gitlab",
+		AllowedMethods:    []string{AppConnectionGitlabAuthMethodAccessToken},
+		CredentialsAttributes: map[string]schema.Attribute{
+			"access_token": schema.StringAttribute{
+				Required:    true,
+				Description: "The GitLab access token for authentication.",
+				Sensitive:   true,
+			},
+			"instance_url": schema.StringAttribute{
+				Required:    true,
+				Description: "The GitLab instance URL (e.g., https://gitlab.com for GitLab.com or your self-hosted GitLab URL).",
+			},
+			"access_token_type": schema.StringAttribute{
+				Required:    true,
+				Description: "The type of access token. Must be 'project' for project access tokens.",
+			},
+		},
+		ReadCredentialsForCreateFromPlan: func(ctx context.Context, plan AppConnectionBaseResourceModel) (map[string]interface{}, diag.Diagnostics) {
+			credentialsConfig := make(map[string]interface{})
+
+			var credentials AppConnectionGitlabCredentialsModel
+			diags := plan.Credentials.As(ctx, &credentials, basetypes.ObjectAsOptions{})
+			if diags.HasError() {
+				return nil, diags
+			}
+
+			if plan.Method.ValueString() != AppConnectionGitlabAuthMethodAccessToken {
+				diags.AddError(
+					"Unable to create GitLab app connection",
+					"Invalid method. Only access-token method is supported. Note: GitLab OAuth connections must be created through the Infisical UI.",
+				)
+				return nil, diags
+			}
+
+			if credentials.AccessTokenType.ValueString() != "project" {
+				diags.AddError(
+					"Unable to create GitLab app connection",
+					"Invalid access_token_type. Only 'project' is supported.",
+				)
+				return nil, diags
```

Comment:
so we don't support the Personal Access Token type?

<img width="801" height="831" alt="image" src="https://github.com/user-attachments/assets/525e7c4f-2f08-4863-88a5-de51c20a05a5" />